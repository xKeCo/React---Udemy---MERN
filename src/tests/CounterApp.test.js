import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas para CounterApp", () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("Debe mostrar el valor por defecto de CounterApp", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el valor por defecto de que se le mande", () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);

    const counterText = wrapper.find("#counter").text().trim();

    expect(counterText).toBe(`${value}`);
  });

  test("Debe de incrementar con el boton de +1", () => {
    wrapper.find("button").at(0).simulate("click");
    const counterText = wrapper.find("#counter").text().trim();

    expect(counterText).toBe("101");
  });

  test("Debe de reducir con el boton de -1", () => {
    wrapper.find("button").at(1).simulate("click");
    const counterText = wrapper.find("#counter").text().trim();

    expect(counterText).toBe("99");
  });

  test("Debe de colocar el valor por defecto con el boton de reset", () => {
    const value = 105;
    const wrapper = shallow(<CounterApp value={value} />);

    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(2).simulate("click");

    const counterText = wrapper.find("#counter").text().trim();

    expect(counterText).toBe(`${value}`);
  });
});
