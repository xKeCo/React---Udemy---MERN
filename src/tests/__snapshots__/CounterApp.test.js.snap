// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para CounterApp Debe mostrar el valor por defecto de CounterApp 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    id="counter"
  >
     
    100
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
